mackerel-plugin-aws-dynamodb/lib/aws-dynamo-db.go

```diff
@@ -0,0 +1,387 @@
+	}}
+	for _, met := range defaultMetricsGroup {
+		dp, err := getLastPointFromCloudWatch(p.CloudWatch, met, tableDimensions)
+		if err == nil {
+			for _, m := range met.Metrics {
+				stats = transformAndAppendDatapoint(dp, m.Type, m.MackerelName, stats)
+			}
+		} else {
+			log.Printf("%s: %s", met, err)
+		}
+	}
+
+	for _, met := range operationalMetricsGroup {
+		operationalStats, err := fetchOperationWildcardMetrics(p.CloudWatch, met, tableDimensions)
+		if err == nil {
+			for name, s := range operationalStats {
+				stats[name] = s
+			}
+		} else {
+			log.Printf("%s: %s", met, err)
+		}
+	}
+	return transformMetrics(stats), nil
+}
+
+// TransformMetrics converts some of datapoints to post differences of two metrics
+func transformMetrics(stats map[string]interface{}) map[string]interface{} {
+	// Although stats are interface{}, those values from cloudwatch.Datapoint are guaranteed to be numerical
+	if consumedReadCapacitySum, ok := stats["ConsumedReadCapacityUnitsSum"].(float64); ok {
+		stats["ConsumedReadCapacityUnitsNormalized"] = consumedReadCapacitySum / 60.0
+	}
+	if consumedWriteCapacitySum, ok := stats["ConsumedWriteCapacityUnitsSum"].(float64); ok {
+		stats["ConsumedWriteCapacityUnitsNormalized"] = consumedWriteCapacitySum / 60.0
+	}
+	return stats
+}
+
+// GraphDefinition of DynamoDBPlugin
+func (p DynamoDBPlugin) GraphDefinition() map[string]mp.Graphs {
+	labelPrefix := strings.Title(p.Prefix)
+	labelPrefix = strings.Replace(labelPrefix, "-", " ", -1)
+
+	var graphdef = map[string]mp.Graphs{
+		"ReadCapacity": {
+			Label: (labelPrefix + " Read Capacity Units"),
+			Unit:  "float",
+			Metrics: []mp.Metrics{
+				{Name: "ProvisionedReadCapacityUnits", Label: "Provisioned"},
+				{Name: "ConsumedReadCapacityUnitsNormalized", Label: "Consumed"},
+				{Name: "ConsumedReadCapacityUnitsAverage", Label: "Consumed (Average per request)"},
+			},
+		},
+		"WriteCapacity": {
+			Label: (labelPrefix + " Write Capacity Units"),
+			Unit:  "float",
+			Metrics: []mp.Metrics{
+				{Name: "ProvisionedWriteCapacityUnits", Label: "Provisioned"},
+				{Name: "ConsumedWriteCapacityUnitsNormalized", Label: "Consumed"},
+				{Name: "ConsumedWriteCapacityUnitsAverage", Label: "Consumed (Average per request)"},
+			},
+		},
+		"ThrottledEvents": {
+			Label: (labelPrefix + " Throttle Events"),
+			Unit:  "integer",
+			Metrics: []mp.Metrics{
+				{Name: "ReadThrottleEvents", Label: "Read"},
+				{Name: "WriteThrottleEvents", Label: "Write"},
+			},
+		},
+		"ConditionalCheckFailedRequests": {
+			Label: (labelPrefix + " ConditionalCheckFailedRequests"),
+			Unit:  "integer",
+			Metrics: []mp.Metrics{
+				{Name: "ConditionalCheckFailedRequests", Label: "Counts"},
+			},
+		},
+		"ThrottledRequests": {
+			Label: (labelPrefix + " ThrottledRequests"),
+			Unit:  "integer",
+			Metrics: []mp.Metrics{
+				{Name: "*", Label: "%1", Stacked: true},
+			},
+		},
+		"SystemErrors": {
+			Label: (labelPrefix + " SystemErrors"),
```

Comment:
For plugins which uses CloudWatch as data sources, IMO using original CloudWatch metric names is better because users easily imagine what those metrics are taken from.